docs/docs/migration-guide.mdx

```diff
@@ -126,6 +126,152 @@ confidences. This change can affect the maximum confidence predicted by the
 However, the previous behavior can still be enforced by setting
 `ranking_length=10` and `renormalize_confidences=True`.
 
+### Slot Mappings
+
+As of version 3.0, there is a single explicit mechanism for slot filling enabled by defining slot mappings for each slot
+in the `slots` section of the domain file. This approach allows slots to be kept up to date over the course of a conversation, and removes duplicated effort in
+mapping the same slots in multiple forms.
+
+This new mechanism replaces the implicit slot setting via auto-fill of slots with entities of the same name.
+The `auto_fill` key in the domain is no longer available, as well as the `auto_fill` parameter in the constructor of
+the `Slot` class.
+
+While forms continue to request the next slot, the extraction of slots is now delegated exclusively to the new default
+action `action_extract_slots`. This action runs in the background after each user turn without it being required to be
+triggered manually.
+
+You are now required to add the `mappings` key for each slot in the 3.0 domain format, including any slots with custom
+mappings. The keys used to define predefined mappings in 2.0 were maintained. Additionally, you can define slots with
+custom mappings implemented in a custom action, like so:
```

Comment:
What about slots that really don't have mappings? e.g. they are just there to be filled in the case that a custom action wants to set them. 


```suggestion
Each slot in the `slots` section must include the `mappings` key. The same keys used for predefined mappings in 2.0 are available in 3.0. Additionally, you can define slots with
custom mappings implemented in a custom action, for example:
```

Reply:
Every slot that gets listed in the domain needs to have the `mappings` parameter defined, as well as everytime a `Slot` class gets instantiated (although I don't have real-life examples for the latter part). There is a little hack I used in the case of default slots internally, (e.g, `session_started_metadata`) meaning I assigned an empty list to mappings argument 😄 But that's not something I advise for domain slots. Thinking about this now, I should actually add a test case for this, when a domain slot has no mapping in `mappings`. I assume it would never get set when `action_extract_slots` runs.

Reply:
To follow-up on this, if a slot doesn't specify mappings (via the domain hack of `mappings: []`), that slot never gets set via a form or `action_extract_slots`.

Reply:
No idea why I only saw this now, but cool! What is the difference between doing `mappings: []` and `mappings: - type: custom` without any custom action to extract the slot?

There should indeed be a test case for empty mappings.

Reply:
There's no difference really, other than that `type: custom` is an explicit way of indicating this slot can be set in other ways than the predefined ways (could be custom component or validation actions etc).